Formula/eget.rb

```diff
@@ -0,0 +1,29 @@
+class Eget < Formula
+  desc "Easily install prebuilt binaries from GitHub"
+  homepage "https://github.com/zyedidia/eget"
+  url "https://github.com/zyedidia/eget/archive/refs/tags/v1.1.0.tar.gz"
+  sha256 "0075a032e31e0525a8171b326ed8fb923e3d1e5e30201557d6820cf2ffd047ee"
+  license "MIT"
+  head "https://github.com/zyedidia/eget.git", branch: "master"
+
+  depends_on "go" => :build
+  depends_on "pandoc" => :build
+
+  def install
+    system "go", "build",
+      *std_go_args(ldflags: "-s -w -X main.Version=#{version}")
+    system "make", "eget.1"
+    man1.install "eget.1"
+  end
+
+  test do
+    # Use eget to install a v1.1.0 release of itself,
+    # and verify that the installed binary is functional.
+    mkdir "test-dir/"
+    system bin/"eget", "zyedidia/eget",
+      "--tag", "v1.1.0",
+      "--to", "test-dir",
+      "--file", "eget"
+    assert_match "eget version 1.1.0", shell_output("test-dir/eget -v")
```

Comment:
```suggestion
    system bin/"eget", "zyedidia/eget",
                       "--tag", "v1.1.0",
                       "--to", testpath,
                       "--file", "eget"
    assert_match "eget version 1.1.0", shell_output("./eget -v")
    assert_match version.to_s, shell_output("#{bin}/eget -v")
```
You probably don't need to create that separate directory, do you? Let's also check that the `main.Version` flag we pass to `go` during the build doesn't go stale.

Reply:
Also, would you mind clarifying what you are looking for when you want to make sure that the "`main.Version` flag we pass to `go` during the build doesn't go stale"? I'm not quite sure what you mean by that. (My bad)

Reply:
> What does line 26 that you added do here? Won't this additional check fail because "1.1.0" != "eget version 1.1.0"?

It's a partial match, "eget version 1.1.0" does contain "1.1.0".

> the "main.Version flag we pass to go during the build doesn't go stale"

If the app changes the config name, version info will stop working. This check ensures that it keeps working.

Reply:
Thank you for clarifying. This suggestion is great (sorry for not understanding as first).

I'm thinking that the check on line 26 should maybe go before the comment on lines 19 and 20, or a blank line should be put between it and the previous lines, that way line 26 is more easily identifiable as a separate test.